### ISSUE TITLE
Is there a way to abort long-running load/decode operations?

### ISSUE THREAD

**Issue by cmbruns** [lilohuang/PyTurboJPEG#78]
Something like

```python
  for status in incremental_decoder(in_file_stream, chunk_size):
     if user_pressed_cancel():
      break
    if status == COMPLETE:
     bgr_array = status.result
     break
```

...preferably with a larger default chunk size than the libjpeg-turbo C `jpeg_read_scanlines(...)` call.

**Comment by lilohuang** [comment]
TL;DR: PyTurboJPEG is a wrapper for the TurboJPEG C-API of the libjpeg-turbo library, which does not support the jpeg_read_scanlines function call or a similar function to abort long-running load/decode operations.

A viable workaround is using Python's multiprocessing library or a similar approach to launch a worker process, and call the terminate() function if it exceeds a timeout value. https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.terminate

The libjpeg-turbo library provides two C-APIs: the TurboJPEG C-API and the libjpeg C-API. The jpeg_read_scanlines function is only available in the conventional libjpeg C-API. Even the official TurboJPEG Java API doesn't provide this functionality. See details at [libjpeg-turbo Documentation](https://libjpeg-turbo.org/Documentation/Documentation).

Since this is outside the scope of using the TurboJPEG C-API, I’m going to close this issue. Thank you for your understanding.
